### ISSUE TITLE
JTReg Test Failure : java/lang/invoke/VarHandles/VarHandleTestMethodHandleAccessXXX.java (2)

### ISSUE THREAD

**Issue by adam-thorpe** [eclipse/openj9#8661]
Failure link
------------
Follow on from: https://github.com/eclipse/openj9/issues/3940
https://ci.adoptopenjdk.net/user/adam-thorpe/my-views/view/OpenJ9%20sanity%20openjdk/job/Test_openjdk11_j9_sanity.openjdk_x86-64_mac/151/console
- test category: sanity.openjdk
- OS/architecture: jdk11+ all platforms

Optional info
-------------
- intermittent failure (yes|no): no
- regression or new test: regression

Failure output (captured from console output)
---------------------------------------------
I can see there's still some conversation and open PR's from the previous issue, I just wanted to raise this if this wasn't already a known issue.

```
00:38:27  Type=Segmentation error vmState=0x000501ff
00:38:27  J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000000
00:38:27  Handler1=000000000228F370 Handler2=0000000002487740
00:38:27  RDI=6E692E656C646E61 RSI=0000000000000003 RAX=0000000000000003 RBX=0000700000BBB160
00:38:27  RCX=0000000000000003 RDX=00000000770E3090 R8=00000000770D0BB0 R9=000000000000000A
00:38:27  R10=00000000037EE921 R11=00006FFFFD3CB63E R12=0000000000007FFF R13=0000000077097F20
00:38:27  R14=0000000000000003 R15=00000000037EE7DC
00:38:27  RIP=0000000002B2B58A GS=0000 FS=0000 RSP=0000700000BB9F88
00:38:27  RFlags=0000000000010202 CS=002B RBP=6E692E656C646E61 ERR=6921A00800000000
00:38:27  TRAPNO=000000000000000D CPU=A008000000000000 FAULTVADDR=00007FFD6921A008
00:38:27  XMM0 000000a5000000c0 (f: 192.000000, d: 3.501293e-312)
00:38:27  XMM1 00ff00ffffffffff (f: 4294967296.000000, d: 7.064164e-304)
00:38:27  XMM2 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:38:27  XMM3 00000000770503a0 (f: 1996817280.000000, d: 9.865588e-315)
00:38:27  XMM4 0000000077050ef0 (f: 1996820224.000000, d: 9.865603e-315)
00:38:27  XMM5 0000000077011df0 (f: 1996561920.000000, d: 9.864326e-315)
00:38:27  XMM6 000000007704f430 (f: 1996813312.000000, d: 9.865569e-315)
00:38:27  XMM7 00000000770503a0 (f: 1996817280.000000, d: 9.865588e-315)
00:38:27  XMM8 0000000077050ef0 (f: 1996820224.000000, d: 9.865603e-315)
00:38:27  XMM9 000000d0000000d0 (f: 208.000000, d: 4.413751e-312)
00:38:27  XMM10 0000002000000020 (f: 32.000000, d: 6.790387e-313)
00:38:27  XMM11 8000000080000000 (f: 2147483648.000000, d: -1.060998e-314)
00:38:27  XMM12 0000002000000020 (f: 32.000000, d: 6.790387e-313)
00:38:27  XMM13 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:38:27  XMM14 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:38:27  XMM15 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:38:27  Module=/Users/jenkins/workspace/Test_openjdk11_j9_sanity.openjdk_x86-64_mac/openjdkbinary/j2sdk-image/Contents/Home/lib/compressedrefs/libj9jit29.dylib
00:38:27  Module_base_address=00000000025F8000 Symbol=_ZN15TR_PrexArgument14knowledgeLevelEPS_
00:38:27  Symbol_address=0000000002B2B580
00:38:27  
00:38:27  Method_being_compiled=java/lang/invoke/BruteArgumentMoverHandle.invokeExact_thunkArchetype_X(I)I
00:38:27  Target=2_90_20200225_487 (Mac OS X 10.14.5)
00:38:27  CPU=amd64 (4 logical CPUs) (0x200000000 RAM)

00:38:27  ----------- Stack Backtrace -----------
00:38:27  ---------------------------------------
00:38:27  JVMDUMP039I Processing dump event "gpf", detail "" at 2020/02/25 16:36:28 - please wait.
00:38:27  JVMDUMP032I JVM requested System dump using '/Users/jenkins/workspace/Test_openjdk11_j9_sanity.openjdk_x86-64_mac/openjdk-tests/TKG/test_output_15826763584317/jdk_lang_0/work/scratch/1/core.20200225.163628.92838.0001.dmp' in response to an event
00:38:27  JVMDUMP012E Error in System dump: The core file created by child process with pid = 92960 was not found. Expected to find core file with name "/cores/core.92960"
00:38:27  JVMDUMP032I JVM requested Java dump using '/Users/jenkins/workspace/Test_openjdk11_j9_sanity.openjdk_x86-64_mac/openjdk-tests/TKG/test_output_15826763584317/jdk_lang_0/work/scratch/1/javacore.20200225.163628.92838.0002.txt' in response to an event

00:38:27  JVMDUMP039I Processing dump event "gpf", detail "" at 2020/02/25 16:36:40 - please wait.
00:38:27  JVMDUMP032I JVM requested System dump using '/Users/jenkins/workspace/Test_openjdk11_j9_sanity.openjdk_x86-64_mac/openjdk-tests/TKG/test_output_15826763584317/jdk_lang_0/work/scratch/1/core.20200225.163640.92838.0003.dmp' in response to an event
00:38:27  JVMDUMP012E Error in System dump: The core file created by child process with pid = 92969 was not found. Expected to find core file with name "/cores/core.92969"
00:38:27  JVMDUMP032I JVM requested Java dump using '/Users/jenkins/workspace/Test_openjdk11_j9_sanity.openjdk_x86-64_mac/openjdk-tests/TKG/test_output_15826763584317/jdk_lang_0/work/scratch/1/javacore.20200225.163640.92838.0004.txt' in response to an event
00:38:27  JVMDUMP010I Java dump written to /Users/jenkins/workspace/Test_openjdk11_j9_sanity.openjdk_x86-64_mac/openjdk-tests/TKG/test_output_15826763584317/jdk_lang_0/work/scratch/1/javacore.20200225.163628.92838.0002.txt
00:38:27  JVMDUMP032I JVM requested Snap dump using '/Users/jenkins/workspace/Test_openjdk11_j9_sanity.openjdk_x86-64_mac/openjdk-tests/TKG/test_output_15826763584317/jdk_lang_0/work/scratch/1/Snap.20200225.163628.92838.0005.trc' in response to an event
00:38:27  JVMDUMP010I Snap dump written to /Users/jenkins/workspace/Test_openjdk11_j9_sanity.openjdk_x86-64_mac/openjdk-tests/TKG/test_output_15826763584317/jdk_lang_0/work/scratch/1/Snap.20200225.163628.92838.0005.trc
00:38:27  JVMDUMP007I JVM Requesting JIT dump using '/Users/jenkins/workspace/Test_openjdk11_j9_sanity.openjdk_x86-64_mac/openjdk-tests/TKG/test_output_15826763584317/jdk_lang_0/work/scratch/1/jitdump.20200225.163628.92838.0006.dmp'
00:38:27  JVMDUMP010I JIT dump written to /Users/jenkins/workspace/Test_openjdk11_j9_sanity.openjdk_x86-64_mac/openjdk-tests/TKG/test_output_15826763584317/jdk_lang_0/work/scratch/1/jitdump.20200225.163628.92838.0006.dmp
00:38:27  JVMDUMP013I Processed dump event "gpf", detail "".
```

**Comment by smlambert** [comment]
excluded via: https://github.com/AdoptOpenJDK/openjdk-tests/pull/1655

**Comment by smlambert** [comment]
Artifacts from Test run 148 (~38.7Mb) and 153 (80.7Mb) copied here: https://ibm.ent.box.com/folder/105135247420 

FIrst occurrence seems to be on Feb 22, not seen in runs prior to Feb 21

Feb 22 (Run 148)
java -version
19:20:00  openjdk version "11.0.7" 2020-04-14
19:20:00  OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.7+3-202002222335)
19:20:00  Eclipse OpenJ9 VM AdoptOpenJDK (build master-79aa8cfe4, JRE 11 Mac OS X amd64-64-Bit Compressed References 20200222_484 (JIT enabled, AOT enabled)
19:20:00  OpenJ9   - 79aa8cfe4
19:20:00  OMR      - 00c57af16
19:20:00  JCL      - 75dba088c4 based on jdk-11.0.7+3)

Feb 21 (Run 147)
java -version
19:19:07  openjdk version "11.0.7" 2020-04-14
19:19:07  OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.7+3-202002212351)
19:19:07  Eclipse OpenJ9 VM AdoptOpenJDK (build master-79aa8cfe4, JRE 11 Mac OS X amd64-64-Bit Compressed References 20200221_483 (JIT enabled, AOT enabled)
19:19:07  OpenJ9   - 79aa8cfe4
19:19:07  OMR      - 00c57af16
19:19:07  JCL      - 75dba088c4 based on jdk-11.0.7+3)

I have had to disable testing on openjdk jdk11 j9 on xLinux do to excessive numbers of crashes and test artifacts as large as 690Mb.  I perhaps will raise this as a separate issue, as it does not seem to be related to a particular test.

**Comment by DanHeidinga** [comment]
@babsingh Is this related to the VarHandle refactoring as part of JEP370?

**Comment by babsingh** [comment]
@smlambert which specific test targets will reproduce the issue? the above test job link is no longer accessible.

**Comment by smlambert** [comment]
It is NOT a specific test or test target that causes this, it appears to be crashing during various times in the test run.  You can unzip one of the artifacts and search for the dmp files.

**Comment by smlambert** [comment]
On an xLinux artifact I looked at there are the following:

./jdk_math_0/work/java/math/BigInteger/DivisionOverflow/core.20200223.003919.14969.0001.dmp
./jdk_util_0/work/java/util/ResourceBundle/Bug4168625Test/core.20200223.020059.3267.0001.dmp
./jdk_lang_0/work/java/lang/ref/SoftReference/Pin/core.20200223.002833.17157.0001.dmp
./jdk_lang_0/work/java/lang/ProcessBuilder/Basic_id1/core.20200223.002728.13138.0001.dmp
./jdk_lang_0/work/java/lang/ProcessBuilder/Basic_id0/core.20200223.002725.12959.0001.dmp
./jdk_lang_0/work/java/lang/ProcessBuilder/Basic_id0/core.20200223.002802.16063.0001.dmp
./jdk_lang_0/work/vm/gc/ArraySize/core.20200223.003816.14239.0001.dmp
./jdk_lang_0/work/vm/gc/InfiniteList/core.20200223.003816.14253.0001.dmp

**Comment by smlambert** [comment]
In the windows artifact:
./jdk_math_0/work/java/math/BigInteger/DivisionOverflow/core.20200226.183920.16492.0001.dmp
./jdk_util_0/work/java/util/ResourceBundle/Bug4168625Test/core.20200226.193858.17740.0001.dmp
./jdk_lang_0/work/java/lang/ref/SoftReference/Pin/core.20200226.182907.23688.0001.dmp
./jdk_lang_0/work/java/lang/invoke/VarHandles/VarHandleTestMethodHandleAccessChar/jitdump.20200226.182443.17864.0004.dmp
./jdk_lang_0/work/java/lang/invoke/VarHandles/VarHandleTestMethodHandleAccessChar/core.20200226.182443.17864.0001.dmp
./jdk_lang_0/work/java/lang/invoke/VarHandles/VarHandleTestMethodHandleAccessBoolean/jitdump.20200226.182351.21688.0004.dmp
./jdk_lang_0/work/java/lang/invoke/VarHandles/VarHandleTestMethodHandleAccessBoolean/core.20200226.182351.21688.0001.dmp
./jdk_lang_0/work/java/lang/invoke/VarHandles/VarHandleTestMethodHandleAccessInt/core.20200226.182522.15024.0001.dmp
./jdk_lang_0/work/java/lang/invoke/VarHandles/VarHandleTestMethodHandleAccessInt/jitdump.20200226.182522.15024.0004.dmp
./jdk_lang_0/work/java/lang/invoke/VarHandles/VarHandleTestMethodHandleAccessLong/jitdump.20200226.182533.18132.0004.dmp
./jdk_lang_0/work/java/lang/invoke/VarHandles/VarHandleTestMethodHandleAccessLong/core.20200226.182533.18132.0001.dmp
./jdk_lang_0/work/java/lang/invoke/VarHandles/VarHandleTestMethodHandleAccessString/jitdump.20200226.182550.22144.0004.dmp
./jdk_lang_0/work/java/lang/invoke/VarHandles/VarHandleTestMethodHandleAccessString/core.20200226.182550.22144.0001.dmp
./jdk_lang_0/work/java/lang/invoke/VarHandles/VarHandleTestMethodHandleAccessByte/jitdump.20200226.182402.29124.0004.dmp
./jdk_lang_0/work/java/lang/invoke/VarHandles/VarHandleTestMethodHandleAccessByte/core.20200226.182402.29124.0001.dmp
./jdk_lang_0/work/java/lang/invoke/VarHandles/VarHandleTestMethodHandleAccessFloat/jitdump.20200226.182452.20568.0004.dmp
./jdk_lang_0/work/java/lang/invoke/VarHandles/VarHandleTestMethodHandleAccessFloat/core.20200226.182452.20568.0001.dmp
./jdk_lang_0/work/java/lang/invoke/VarHandles/VarHandleTestMethodHandleAccessDouble/jitdump.20200226.182430.696.0007.dmp
./jdk_lang_0/work/java/lang/invoke/VarHandles/VarHandleTestMethodHandleAccessDouble/core.20200226.182430.696.0002.dmp
./jdk_lang_0/work/java/lang/invoke/VarHandles/VarHandleTestMethodHandleAccessDouble/core.20200226.182430.696.0001.dmp
./jdk_lang_0/work/java/lang/ProcessBuilder/Basic_id0/core.20200226.182826.28924.0001.dmp
./jdk_lang_0/work/java/lang/ProcessBuilder/Basic_id0/core.20200226.182934.30292.0001.dmp
./jdk_lang_0/work/vm/gc/ArraySize/core.20200226.183829.25520.0001.dmp
./jdk_lang_0/work/vm/gc/InfiniteList/core.20200226.183830.20208.0001.dmp

**Comment by pshipton** [comment]
The "good" and "bad" JVMs in https://github.com/eclipse/openj9/issues/8661#issuecomment-592667113 have exactly the same SHAs.

**Comment by babsingh** [comment]
The failure happens during method compilation in all core dumps. Stacktrace mostly consists of JIT frames. ++ @andrewcraik @cathyzhyi.
```
Method_being_compiled=java/lang/invoke/BruteArgumentMoverHandle.invokeExact_thunkArchetype_X(I)I
```

**Comment by babsingh** [comment]
> @babsingh Is this related to the VarHandle refactoring as part of JEP370?

The VarHandle changes are only Java changes, and they only impact Java 14+. I don't think they are responsible for these failures, which are also seen on Java 11.

**Delivered VarHandle changes:** https://github.com/eclipse/openj9/pull/8330 **(Part 1)** and https://github.com/eclipse/openj9/pull/8414 **(Part 2)**.

**Comment by smlambert** [comment]
re: https://github.com/eclipse/openj9/issues/8661#issuecomment-592721396 yes, apologies that report on the mac failures is a red herring.  I have created https://github.com/eclipse/openj9/issues/8694 and will update it more thoroughly.

**Comment by pshipton** [comment]
I think this is fixed now by #8598

**Comment by adam-thorpe** [comment]
Yep seems to be resolved. Tests will be unexcluded
